### PR TITLE
enable logout by setting property

### PR DESCRIPTION
**Pull request by ahmad2x4**
This will fix issue #16.  You can set the logout property and this will force to log out first and log in again.

### CODE REVIEW

File: src/ADLoginView.js [hunk 1]
```diff
@@ -63,8 +63,16 @@ export default class ADLoginView extends React.Component {
     log.debug('ADLoginView updated.')
   }
 
-  componentDidMount() {
-
+  componentWillReceiveProps(nextProps){
+    if(!this.props.needLogout && nextProps.needLogout) {
```
Comment (ahmad2x4) [inline src/ADLoginView.js:67]: Checking needLogout to see if it is changed from false to true.

File: src/ADLoginView.js [hunk 2]
```diff
@@ -63,8 +63,16 @@ export default class ADLoginView extends React.Component {
...
+      let context = this.props.context
+      let tenant = context.getConfig().tenant
+      this._needRedirect = nextProps.needLogout || false
+      this.setState({
```
Comment (ahmad2x4) [inline src/ADLoginView.js:71]: setting the state to visible also getting URL which in this case will be logout

File: src/ADLoginView.js [hunk 3]
```diff
@@ -85,7 +93,7 @@ export default class ADLoginView extends React.Component {
           javaScriptEnabled={true}
           domStorageEnabled={true}
           onLoadEnd={()=>{
-            if(this.props.needLogout){
+            if(this._needRedirect){
```
Comment (ahmad2x4) [inline src/ADLoginView.js:96]: _needRedirect is more accurate flag to check

**Comment by wkh237** [comment]
@ahmad2x4 , thanks for the PR 👍

**Pull request merged**